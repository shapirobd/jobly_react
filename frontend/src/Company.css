.Company {
	margin: 20px 0;
	box-shadow: 0px 2px 2px rgb(131, 131, 131);
	padding: 30px;
}

.Company-link {
	text-decoration: none;
}

.Company:hover {
	box-shadow: 0px 5px 5px rgb(131, 131, 131);
	color: rgb(64, 80, 181);
}

.Company-header {
	display: flex;
	justify-content: space-between;
}

.Company-name {
	display: inline-block;
}

.Company-logo {
	display: inline;
	width: 50px;
	height: auto;
}

.Company-description {
	text-align: left;
	padding-top: 10px;
}
