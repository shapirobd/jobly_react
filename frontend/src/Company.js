import React from "react";
import { Link } from "react-router-dom";
import { Paper, Typography } from "@material-ui/core";
import "./Company.css";

const Company = ({ name, description, logoUrl, handle }) => {
	return (
		<Link to={`/companies/${handle}`} className="Company-link">
			<Paper elevation={3} className="Company">
				<div className="Company-header">
					<Typography className="Company-name" variant="body1">
						{name}
					</Typography>
					{logoUrl ? (
						<img className="Company-logo" src={`.${logoUrl}`} alt={logoUrl} />
					) : null}
				</div>
				<Typography className="Company-description" variant="body2">
					{description}
				</Typography>
			</Paper>
		</Link>
	);
};

export default Company;
