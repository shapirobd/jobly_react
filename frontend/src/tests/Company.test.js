import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Company from "../Company";

test("smoke test", () => {
	render(
		<MemoryRouter>
			<Company />
		</MemoryRouter>
	);
});

test("snapshot test", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<Company />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
