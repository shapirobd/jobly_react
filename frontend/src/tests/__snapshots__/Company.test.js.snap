// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <a
    class="Company-link"
    href="/companies/undefined"
  >
    <div
      class="MuiPaper-root Company MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        class="Company-header"
      >
        <p
          class="MuiTypography-root Company-name MuiTypography-body1"
        />
      </div>
      <p
        class="MuiTypography-root Company-description MuiTypography-body2"
      />
    </div>
  </a>
</DocumentFragment>
`;
